### PR TITLE
Zoma/move

### PR DESCRIPTION
**Pull request by coldplaying42**
Add the integrate work item into chrimc/move branch

### CODE REVIEW

**Comment by chrimc62** [comment]
> 

Delete file

---
Refers to: experimental/rush.json:16 in 6e6e47d. [](commit_id = 6e6e47d92cbd4c7b552b6a8bca233d721e7716ca, deletion_comment = False)

**Comment by chrimc62** [comment]
> dependencies:

Delete file

---
Refers to: experimental/common/config/rush/pnpm-lock.yaml:1 in 6e6e47d. [](commit_id = 6e6e47d92cbd4c7b552b6a8bca233d721e7716ca, deletion_comment = False)

File: experimental/common/scripts/install-run-rush.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+"use strict";
```
Comment (chrimc62) [inline experimental/common/scripts/install-run-rush.js:1]: Delete file

File: experimental/common/scripts/install-run-rushx.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"use strict";
```
Comment (chrimc62) [inline experimental/common/scripts/install-run-rushx.js:1]: Delete file

File: experimental/common/scripts/install-run.js [hunk 3]
```diff
@@ -0,0 +1,433 @@
+"use strict";
```
Comment (chrimc62) [inline experimental/common/scripts/install-run.js:1]: Delete file

File: experimental/generation/generator/package.json [hunk 4]
```diff
@@ -49,7 +50,7 @@
     "rimraf": "^3.0.2",
     "ts-node": "^8.8.2",
     "tslint": "^5.18.0",
-    "typescript": "^3.8.3"
+    "typescript": "3.5.3"
```
Comment (chrimc62) [inline experimental/generation/generator/package.json:53]: Don't roll this back.

File: experimental/generation/generator/tsconfig.json [hunk 5]
```diff
@@ -6,7 +6,8 @@
     "outDir": "lib",
     "rootDir": "src",
     "strict": true,
-    "target": "es2017"
+    "target": "es2017",
+    "noImplicitAny": false
```
Comment (chrimc62) [inline experimental/generation/generator/tsconfig.json:10]: Did you mean to add this?

File: experimental/generation/generator/.rush/temp/package-deps_build.json [hunk 6]
```diff
@@ -0,0 +1,469 @@
+{
+  "files": {
```
Comment (chrimc62) [inline experimental/generation/generator/.rush/temp/package-deps_build.json:2]: Delete file

File: experimental/generation/generator/.rush/temp/shrinkwrap-deps.json [hunk 7]
```diff
@@ -0,0 +1,527 @@
+{
+  "@azure/cognitiveservices-luis-authoring@4.0.0-preview.1": "sha512-HAhnf+57iHn1s7U5H7Rr51JZQINeJeT0uxtXr/Ksa6wbmZDOm+VApWnFwp/+QH1ZnW8S6Qf0roKtuTeEBjmBZA==",
+  "@azure/ms-rest-azure-js@2.0.1": "sha512-5e+A710O7gRFISoV4KI/ZyLQbKmjXxQZ1L8Z/sx7jSUQqmswjTnN4yyIZxs5JzfLVkobU0rXxbi5/LVzaI8QXQ==",
```
Comment (chrimc62) [inline experimental/generation/generator/.rush/temp/shrinkwrap-deps.json:3]: Delete file

File: experimental/generation/generator/src/commands/dialog/generate.ts [hunk 8]
```diff
@@ -26,25 +29,70 @@ export default class DialogGenerate extends Command {
...
+                        } else if (type === gen.FeedbackType.error) {
+                            this.errorMsg(msg)
+                        }
+                    })
+            } catch (e) {
+                this.thrownError(e)
+            }
+
+            try {
+                await integ.integrateAssets(schemaName, tempOldPath, tempNewPath, flags.output, 'en-us', (type, msg) => {
```
Comment (chrimc62) [inline experimental/generation/generator/src/commands/dialog/generate.ts:64]: >'en-us' [](start = 96, length = 7)

Should be flags.locale.

File: experimental/generation/generator/src/commands/dialog/integrate.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
```
Comment (chrimc62) [inline experimental/generation/generator/src/commands/dialog/integrate.ts:4]: Do we still need this command?  I'm wondering when someone would want to integrate, but not regenerate.

File: experimental/generation/generator/src/library/dialogGenerator.ts [hunk 10]
```diff
@@ -78,7 +78,7 @@ export async function isUnchanged(path: string): Promise<boolean> {
     return result
 }
 
-async function writeFile(path: string, val: any, feedback: Feedback) {
+export async function writeFile(path: string, val: any, feedback: Feedback) {
```
Comment (chrimc62) [inline experimental/generation/generator/src/library/dialogGenerator.ts:81]: >writeFile [](start = 22, length = 9)

do you use this function?

File: experimental/generation/generator/src/library/integration.ts [hunk 11]
```diff
@@ -0,0 +1,734 @@
...
+        let oldHash = jsonOld.$Generator
+        let jsonNew = JSON.parse(newFile)
+        let newHash = jsonNew.$Generator
+        if (oldHash && newHash) {
+            result = oldHash === newHash
+        }
+    }
+    return result
+}
+
```
Comment (chrimc62) [inline experimental/generation/generator/src/library/integration.ts:61]: Should use the one from dialogGenerator.  You already import as gen.

File: experimental/generation/generator/src/library/dialogGenerator.ts (hunk 10 above)
Comment (coldplaying42) [reply experimental/generation/generator/src/library/dialogGenerator.ts:81]: Yes, I use this function in integration.ts

File: experimental/generation/generator/src/library/integration.ts (hunk 11 above)
Comment (coldplaying42) [reply experimental/generation/generator/src/library/integration.ts:61]: This isUnchanged function is kind of different from that of dialogGenerator.

File: experimental/generation/generator/src/commands/dialog/integrate.ts (hunk 9 above)
Comment (coldplaying42) [reply experimental/generation/generator/src/commands/dialog/integrate.ts:4]: I'd like to keep the integrate command. I think sometimes user might need to merge two existing assets, and this command could be useful.

File: experimental/generation/generator/tsconfig.json (hunk 5 above)
Comment (coldplaying42) [reply experimental/generation/generator/tsconfig.json:10]: Yes

**Comment by chrimc62** [comment]
Once you get the test to pass, you can check-in.

**Pull request merged**